src/data-workspace/data-entry-cell/entry-field-input.js

```diff
@@ -63,56 +57,21 @@ export function EntryFieldInput({
     categoryOptionCombo: coc,
     setSyncStatus,
     disabled,
+    locked,
 }) {
-    const setHighlightedFieldId = useSetHighlightedFieldIdContext()
-
-    // used so we don't consume the "id" which
-    // would cause this component to rerender
-    const setRightHandPanel = useSetRightHandPanel()
-
     const { id: deId } = de
     const { id: cocId } = coc
     const dataValueParams = useDataValueParams({ deId, cocId })
 
-    const onKeyDown = useCallback(
-        (event) => {
-            const { key, ctrlKey, metaKey } = event
-            const ctrlXorMetaKey = ctrlKey ^ metaKey
-
-            if (ctrlXorMetaKey && key === 'Enter') {
-                setRightHandPanel(dataDetailsSidebarId)
-            } else if (key === 'ArrowDown' || key === 'Enter') {
-                event.preventDefault()
-                focusNext()
-            } else if (key === 'ArrowUp') {
-                event.preventDefault()
-                focusPrev()
-            }
-        },
-        [setRightHandPanel]
-    )
-
-    const onFocus = useCallback(() => {
-        setHighlightedFieldId({ de, coc })
-    }, [de, coc, setHighlightedFieldId])
-
     const sharedProps = useMemo(
         () => ({
             fieldname,
             dataValueParams,
-            disabled,
+            disabled: disabled,
+            locked,
```

Comment:
I thought it would be easier to pass the `locked` boolean to each of the possible inputs from here rather than get it from the context, but that choice was fairly arbitrary